.github/workflows/s3-mirror.yml

```diff
@@ -0,0 +1,41 @@
+name: Mirror repository content to an S3 bucket
+
+on:
+  push:
+    branches:
+      - gh-pages
+
+  schedule:
+    # Pushes to `gh-pages` done from other actions cannot trigger this action so we also want it to run
+    # on a schedule. Let's give the nightly job a 1h head-start and run every day at 1:00.
+    - cron: '0 1 * * *'
+
+env:
+  S3_BUCKET: solc-bin
+  S3_REGION: eu-central-1
+
+jobs:
+  push-to-s3:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    steps:
+      - name: Wait for other instances of this workflow to finish
+        # It's not safe to run two S3 sync operations concurrently with different files
+        uses: softprops/turnstyle@v1
+        with:
+          same-branch-only: no
```

Comment:
Unfortunately there's no way to tell github to run only one instance of the action at a time. As a workaround I used the [turnstyle](https://github.com/softprops/turnstyle) action.

Is it OK to use it? I looked at its code briefly and I didn't see anything nefarious. The docs say that giving it `GITHUB_TOKEN` is required but seems to work without it just fine. So this PR should be completely safe (it can't access secrets) but it's a third-party action so if we ever update it without reviewing the new code **and** we start putting any secrets (`GITHUB_TOKEN` or S3 keys) in env variables it could theoretically steal them.